qcodes/utils/threading.py

```diff
@@ -90,3 +101,87 @@ def thread_map(
         t.start()
 
     return [t.output() for t in threads]
+
+
+class _ParamCaller:
+
+    def __init__(self, *parameters: _BaseParameter):
+
+        self._parameters = parameters
+
+    def __call__(self) -> Tuple[Tuple[_BaseParameter, ParamDataType], ...]:
+        output = []
+        for param in self._parameters:
+            output.append((param, param.get()))
+        return tuple(output)
+
+    def __repr__(self) -> str:
+        names = tuple(param.full_name for param in self._parameters)
+        return f"ParamCaller of {names}"
+
+
+def _instrument_to_param(
+        params: Sequence[ParamMeasT]
+) -> Dict[Optional[str], Tuple[_BaseParameter, ...]]:
+
+    real_parameters = [param for param in params
+                       if isinstance(param, _BaseParameter)]
+
+    output: Dict[Optional[str], Tuple[_BaseParameter, ...]] = defaultdict(tuple)
+    for param in real_parameters:
+        if param.underlying_instrument:
+            output[param.underlying_instrument.full_name] += (param,)
+        else:
+            output[None] += (param,)
+
+    return output
+
+
+def call_params_threaded(param_meas: Sequence[ParamMeasT]) -> OutType:
```

Comment:
let's adda docstring to this now-public function? :)

Reply:
added :)